web/src/components/toolbar/filters/filter-search-input.tsx

```diff
@@ -0,0 +1,619 @@
+import {
+  ActionGroup,
+  Button,
+  Flex,
+  FlexItem,
+  Form,
+  FormGroup,
+  Menu,
+  MenuContent,
+  MenuItem,
+  MenuList,
+  Panel,
+  PanelMain,
+  PanelMainBody,
+  Popper,
+  SearchInput,
+  ValidatedOptions
+} from '@patternfly/react-core';
+import _ from 'lodash';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  FilterCompare,
+  FilterDefinition,
+  FilterOption,
+  Filters,
+  FilterValue,
+  findFromFilters,
+  getCompareText
+} from '../../../model/filters';
+import { getHTTPErrorDetails } from '../../../utils/errors';
+import { matcher } from '../../../utils/filter-definitions';
+import { Indicator, setTargeteableFilterDefinition } from '../../../utils/filters-helper';
+import { useOutsideClickEvent } from '../../../utils/outside-hook';
+import { usePrevious } from '../../../utils/previous-hook';
+import { Direction } from '../filters-toolbar';
+import AutocompleteFilter from './autocomplete-filter';
+import CompareFilter from './compare-filter';
+import { FilterHints } from './filter-hints';
+import './filter-search-input.css';
+import FiltersDropdown from './filters-dropdown';
+import TextFilter from './text-filter';
+
+interface FormUpdateResult {
+  def?: FilterDefinition;
+  comparator?: FilterCompare;
+  value?: string;
+  hasError: boolean;
+}
+
+interface Suggestion {
+  display?: string;
+  value: string;
+  validate?: boolean;
+}
+
+export interface FilterSearchInputProps {
+  filterDefinitions: FilterDefinition[];
+  filters?: Filters;
+  searchInputValue: string;
+  indicator: Indicator;
+  direction: Direction;
+  filter: FilterDefinition;
+  compare: FilterCompare;
+  value: string;
+  setValue: (v: string) => void;
+  setCompare: (v: FilterCompare) => void;
+  setFilter: (v: FilterDefinition) => void;
+  setDirection: (v: Direction) => void;
+  setIndicator: (v: Indicator) => void;
+  setSearchInputValue: (v: string) => void;
+  setFilters: (v: Filters) => void;
+  setMessage: (m: string | undefined) => void;
+}
+
+export const FilterSearchInput: React.FC<FilterSearchInputProps> = ({
```

Comment:
That's quite big, any good opportunity to break it down in smaller components? Like perhaps a parent "FilterSearchInput", and children "FilterSearchSuggestions" and "FilterSearchPanel" ? If that makes sense.